engine/class_modules/priest/sc_priest_shadow.cpp

```diff
@@ -1951,13 +1951,18 @@ void priest_t::generate_apl_shadow()
                       "cooldown.shadow_word_death.remains<gcd.max)" );
   single->add_talent( this, "Surrender to Madness", "if=buff.voidform.stack>10+(10*buff.bloodlust.up)" );
   single->add_talent( this, "Dark Void", "if=raid_event.adds.in>10" );
-  single->add_talent( this, "Mindbender" );
+  single->add_talent( this, "Mindbender", "if=talent.mindbender.enabled|(buff.voidform.stack>18|target.time_to_die<15)" );
   single->add_talent( this, "Shadow Word: Death",
                       "if=!buff.voidform.up|"
                       "(cooldown.shadow_word_death.charges=2&"
                       "buff.voidform.stack<15)" );
   single->add_talent( this, "Shadow Crash", "if=raid_event.adds.in>5&raid_event.adds.duration<20" );
-  single->add_action( this, "Mind Blast", "if=variable.dots_up" );
+  // Bank the Shadow Word: Void charges for a bit to try and avoid overcapping on Insanity.
+  single->add_action( this, "Mind Blast",
```

Comment:
Can you add the same `movement.in` block to the module? This needs to be here so we generate the proper APL.